x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownAndCombineFilters.java

```diff
@@ -38,18 +36,11 @@ protected LogicalPlan rule(Filter filter) {
         LogicalPlan child = filter.child();
         Expression condition = filter.condition();
 
+        // TODO: Push down past STATS if the filter is only on the groups; but take into account that `STATS ... BY field` performs an
+        // implicit `MV_EXPAND field`, which we cannot just push past.
```

Comment:
Please add latest findings here, ie. that it performs and implicit MV_DEDUPE + MV_EXPAND

Reply:
Done!